app/src/org/commcare/google/services/analytics/CCPerfMonitoring.kt

```diff
@@ -0,0 +1,31 @@
+package org.commcare.google.services.analytics
+
+import com.google.firebase.perf.FirebasePerformance
+import com.google.firebase.perf.metrics.Trace
+import org.commcare.android.logging.ReportingUtils
+
+object CCPerfMonitoring {
+    // Traces
+    // Measures the duration of synchronous case list loading
+    const val TRACE_SYNC_ENTITY_LIST_LOADING = "sync_case_list_loading"
+
+    // Attributes
+    const val ATTR_NUM_CASES_LOADED = "number_of_cases_loaded"
+
+    fun startTracing (traceName: String): Trace? {
+        try {
+            val trace = FirebasePerformance.getInstance().newTrace(traceName)
+            trace.putAttribute(CCAnalyticsParam.CCHQ_DOMAIN, ReportingUtils.getDomain())
+            trace.putAttribute(CCAnalyticsParam.USERNAME, ReportingUtils.getUser())
```

Comment:
Actually i think we should be using the same list of attributes as we send to analytics user properties here, think app id is definitely a big one that we should be including here.